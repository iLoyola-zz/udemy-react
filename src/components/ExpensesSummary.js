import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import { Link } from 'react-router-dom';

import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = ( {expenseCount, expensesTotal} ) => {
	const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
	const formattedExpensesTotal = numeral( expensesTotal / 100 ).format( '$0,0.00' );

	return (
		<div>
			<h1>Viewing {expenseCount} {expenseWord} totalling {formattedExpensesTotal}</h1>
			<Link to="/create">Add expense</Link>
		</div>
	)
};

const mapStateToProps = ( state ) => {
	const visibleExpenses = selectExpenses( state.expenses, state.filters );
	return {
		expenseCount: visibleExpenses.length,
		expensesTotal: selectExpensesTotal( visibleExpenses )
	}
};

export default connect( mapStateToProps )( ExpensesSummary );