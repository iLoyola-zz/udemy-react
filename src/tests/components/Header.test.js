import React from 'react';
import { shallow } from 'enzyme';

import { Header } from '../../components/Header';

test( 'Should render Header', () => {
	const wrapper = shallow( <Header startLogout={() => {}} /> );
	expect( wrapper ).toMatchSnapshot();
});

test( 'Should call logout on button click', () => {
	const onLogoutSpy = jest.fn();
	const wrapper = shallow( <Header startLogout={onLogoutSpy} /> );
	wrapper.find( 'button' ).simulate( 'click' );
	expect( onLogoutSpy ).toHaveBeenCalled();
});