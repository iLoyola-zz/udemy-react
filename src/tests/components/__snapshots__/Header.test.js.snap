// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Header 1`] = `
<header>
  <Link
    replace={false}
    to="/dashboard"
  >
    <h1>
      Expensify
    </h1>
  </Link>
  <button
    onClick={[Function]}
  >
    Logout
  </button>
</header>
`;
